src/store/modules/product/getters.ts

```diff
@@ -0,0 +1,10 @@
+import { GetterTree } from 'vuex'
+import ProductState from './ProductState'
+import RootState from '../../RootState'
+
+const getters: GetterTree <ProductState, RootState> = {
+  getProduct: (state) => (productId: string) => {
+    return state.cached[productId] ? state.cached[productId] : 0
```

Comment:
The getProduct will return an empty object in case of no product is found with a specific productId.
```suggestion
    return state.cached[productId] ? state.cached[productId] : {}
```

Reply:
Improved the code.
Thank you sir for reviewing my code.